services/121-service/src/payments/fsp-integration/intersolve-visa/intersolve-visa.service.ts

```diff
@@ -293,6 +285,51 @@ export class IntersolveVisaService
     return returnData;
   }
 
+  // TODO: Remove referenceId and programId from input params.
+  public async getRetrievedAndUpdatedWallet(
+    registrationId: number,
+  ): Promise<IntersolveVisaWalletDto> {
+    const customer =
+      await this.intersolveVisaCustomerScopedRepository.findOneWithWalletsByRegistrationId(
+        registrationId,
+      );
+
+    if (!customer) {
+      throw new HttpException(
+        {
+          errors: `No Customer Entity found for Registration: ${registrationId}`,
+        },
+        HttpStatus.NOT_FOUND,
+      );
+    }
+
+    if (!customer.intersolveVisaParentWallet) {
+      throw new HttpException(
+        {
+          errors: `No ParentWallet Entity found for Registration: ${registrationId}`,
+        },
+        HttpStatus.NOT_FOUND,
+      );
+    }
+
+    // Retrieve and update the parent wallet from Intersolve
+    const intersolveVisaParentWallet = await this.retrieveAndUpdateParentWallet(
+      customer.intersolveVisaParentWallet,
+    );
+
+    // Retrieve and update the current wallet from Intersolve (= skip all substituted wallets, alernatively, sort for newest wallet and take that one)
```

Comment:
These are some nice api calls to skip. :) I think the `alernatively, sort for newest wallet and take that one)` comment is not needed, since the subsitute filter approach is better?